test/baggage/baggage.py

```diff
@@ -106,10 +106,13 @@ def to_string(self) -> str:
 class BaggageEntryProperty(object):
     def __init__(self, key: str, value: str = None) -> None:
         self.key = key
-        self.value = value
+        if value is not None:
+          self.value = unquote(value)
+        else:
+          self.value = value
```

Comment:
Note: The reference implementation actually did not encode or decode property values until now (only baggage entry values have been encoded/decoded). This behavior needs to be updated to be still in line with the spec wording.